src/main/java/org/folio/auth/authtokenmodule/LimitedSizeQueue.java

```diff
@@ -0,0 +1,41 @@
+package org.folio.auth.authtokenmodule;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LimitedSizeQueue<K>
+    extends LinkedHashMap<K, Boolean> {
+  private final int maxSize;
+
+  public LimitedSizeQueue(int size) {
+    maxSize = size;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!super.equals(obj)) {
+      return false;
+    }
+    LimitedSizeQueue<K> fobj = (LimitedSizeQueue<K>) obj;
+    return fobj.maxSize == maxSize;
+  }
+
+  @Override
+  public int hashCode() {
+    return super.hashCode();
```

Comment:
```suggestion
    return super.hashCode() + 31 * maxSize;
```
Because maxSize is relevant for equals.

Reply:
ok